### ISSUE TITLE
False positive for empty_line_after_outer_attr with macro

### ISSUE THREAD

**Issue by cesarb** [rust-lang-nursery/rust-clippy#2453]
The following input:

```rust
macro_rules! decl_simd {
    ($($decl:item)*) => {
        $(
            #[derive(Clone, Copy, Debug)]
            #[repr(C)]
            $decl
        )*
    }
}

decl_simd! {
    pub struct Simd2<T>(pub T, pub T);
    pub struct Simd4<T>(pub T, pub T, pub T, pub T);
    pub struct Simd8<T>(pub T, pub T, pub T, pub T,
                        pub T, pub T, pub T, pub T);
    pub struct Simd16<T>(pub T, pub T, pub T, pub T,
                         pub T, pub T, pub T, pub T,
                         pub T, pub T, pub T, pub T,
                         pub T, pub T, pub T, pub T);
    pub struct Simd32<T>(pub T, pub T, pub T, pub T,
                         pub T, pub T, pub T, pub T,
                         pub T, pub T, pub T, pub T,
                         pub T, pub T, pub T, pub T,
                         pub T, pub T, pub T, pub T,
                         pub T, pub T, pub T, pub T,
                         pub T, pub T, pub T, pub T,
                         pub T, pub T, pub T, pub T);
}
```

Results in the following false positive warnings:

```
warning: Found an empty line after an outer attribute. Perhaps you forgot to add a '!' to make it an inner attribute?
  --> src/lib.rs:5:13
   |
5  | /             #[repr(C)]
6  | |             $decl
7  | |         )*
8  | |     }
...  |
11 | | decl_simd! {
12 | |     pub struct Simd2<T>(pub T, pub T);
   | |____^
   |
   = note: #[warn(empty_line_after_outer_attr)] on by default
   = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.186/index.html#empty_line_after_outer_attr

warning: Found an empty line after an outer attribute. Perhaps you forgot to add a '!' to make it an inner attribute?
  --> src/lib.rs:5:13
   |
5  | /             #[repr(C)]
6  | |             $decl
7  | |         )*
8  | |     }
...  |
12 | |     pub struct Simd2<T>(pub T, pub T);
13 | |     pub struct Simd4<T>(pub T, pub T, pub T, pub T);
   | |____^
   |
   = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.186/index.html#empty_line_after_outer_attr

warning: Found an empty line after an outer attribute. Perhaps you forgot to add a '!' to make it an inner attribute?
  --> src/lib.rs:5:13
   |
5  | /             #[repr(C)]
6  | |             $decl
7  | |         )*
8  | |     }
...  |
13 | |     pub struct Simd4<T>(pub T, pub T, pub T, pub T);
14 | |     pub struct Simd8<T>(pub T, pub T, pub T, pub T,
   | |____^
   |
   = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.186/index.html#empty_line_after_outer_attr

warning: Found an empty line after an outer attribute. Perhaps you forgot to add a '!' to make it an inner attribute?
  --> src/lib.rs:5:13
   |
5  | /             #[repr(C)]
6  | |             $decl
7  | |         )*
8  | |     }
...  |
15 | |                         pub T, pub T, pub T, pub T);
16 | |     pub struct Simd16<T>(pub T, pub T, pub T, pub T,
   | |____^
   |
   = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.186/index.html#empty_line_after_outer_attr

warning: Found an empty line after an outer attribute. Perhaps you forgot to add a '!' to make it an inner attribute?
  --> src/lib.rs:5:13
   |
5  | /             #[repr(C)]
6  | |             $decl
7  | |         )*
8  | |     }
...  |
19 | |                          pub T, pub T, pub T, pub T);
20 | |     pub struct Simd32<T>(pub T, pub T, pub T, pub T,
   | |____^
   |
   = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.186/index.html#empty_line_after_outer_attr

```

**Comment by oli-obk** [comment]
cc @phansch

**Comment by phansch** [comment]
I will take a look at it today/tomorrow 👍

**Comment by phansch** [comment]
@oli-obk I managed to reproduce it, just not sure what the expected outcome should be.
Do you think we should ignore the `#[repr(C)]`, too, as it is inside the macro definition? Currently `in_macro()` doesn't seem to catch this because there is no expn_info. However, there is a macro_backtrace in the span, so it should be possible to skip this attribute.

**Comment by phansch** [comment]
@cesarb Does this still happen with the latest clippy release? I can't seem to reproduce this anymore.

**Comment by phansch** [comment]
Going ahead and closing this. Feel free to open a new issue if this still happens with the latest release :heart:
